docker/libra-build.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/sh
+# Copyright (c) The Libra Core Contributors
+# SPDX-License-Identifier: Apache-2.0
+set -e
+
+DOCKERFILE=$1
+TAG=$2
+
+if [ -z "$DOCKERFILE" ] || [ -z "$TAG" ]
+then
+  echo "Usage libra-build.sh <Docker_file> <Local_tag> <Other_args>"
+fi
+
+shift 2
+
+DIR="$( cd "$( dirname "$0" )" && pwd )"
+
+PROXY=""
+if [ "$https_proxy" ]; then
+    PROXY=" --build-arg https_proxy=$https_proxy --build-arg http_proxy=$http_proxy"
+fi
+
+for _ in seq 1 30; do
```

Comment:
Is it possible to get detailed docker build failure status? If we can do that, I think we can retry based on specific build failure.  For example, it's worth retrying if the build failed due to flaky network or external infra. If the build failed due to compilation issue in Rust and 3rd party tooling, it is probably better to stop right away.

Reply:
Heads up - the build step in LBT also has retry logic but for different failure mechanism (https://github.com/libra/libra/blob/ba042341101cfec720f6ec46992bf0bb3c99bbb1/.github/workflows/land-blocking.yml#L35-L36)

You might want to take a look and adjust the retry limit as needed.

Reply:
I am not sure how to separate build failure between retriable and non-retriable. If we have ideas we can do it in next PR.

Maybe I should just delete retry from docker - to be completely honest I am not sure what does it help with(crates.io issues? How often are they? are they solved within retry period?)

LBT need separate retry regardless of docker builds, because most common failure there happens before even getting to docker compilation